batch/batch/worker/worker.py

```diff
@@ -408,33 +415,39 @@ def __init__(self, job, name, spec):
         self.overlay_path = None
 
         self.image_config = None
+        self.image_digest = None
         self.rootfs_path = None
         scratch = self.spec['scratch']
         self.container_scratch = f'{scratch}/{self.name}'
         self.container_overlay_path = f'{self.container_scratch}/rootfs_overlay'
         self.config_path = f'{self.container_scratch}/config'
 
+        self.container_name = f'batch-{self.job.batch_id}-job-{self.job.job_id}-{self.name}'
+
         self.netns: Optional[NetworkNamespace] = None
         self.process = None
 
-    async def run(self, worker):
+    async def run(self, worker: 'Worker'):
         try:
 
             async def localize_rootfs():
-                async with worker.rootfs_locks[self.image_ref_str]:
+                cond = worker.rootfs_locks[self.image_ref_str]
```

Comment:
Should `worker.rootfs_locks` be `worker.rootfs_conds`? It's a bit confusing since a lock is also an asyncio object. I understand though you're using this as a lock.

Reply:
So if two jobs with the same image are waiting for the 10th image to delete, what happens? Do they each have to wait for an image to be cleared or does one break when the other job has cleared the last image and only one image is deleted?

Reply:
Ya that's not ideal. If two jobs are waiting on the same image, the first will delete an unused image and populate the new one, and the second will delete a second unused image and not repopulate, so at the end we'll end up with 9 images in the cache. I suppose I can change this condition to be `if cache-is-full and need-to-pull-image`

Reply:
also yes I will rename to `rootfs_conds`.

Reply:
Ended up reverting this to just normal locks again and have a separate GC loop for removing unused images.